### PR TITLE
fix: upgrade to latest kin-openapi [try-2]

### PR DESCRIPTION
**Pull request by tinygrasshopper**

### CODE REVIEW

File: remove_elements.go [hunk 1]
```diff
@@ -65,45 +63,31 @@ func RemoveElements(doc *openapi3.T, excludes ExcludePatterns) error {
...
-// StructField implements reflectwalk.StructWalker.
-func (ex *excluder) StructField(field reflect.StructField, v reflect.Value) error {
-	if field.Name != "Extensions" || !v.CanInterface() {
-		return nil
-	}
-
-	switch v.Interface().(type) {
-	case map[string]interface{}:
-		ex.applyExtensions(v.Addr().Interface().(*map[string]interface{}))
+	for _, pathItem := range ex.doc.Paths.Map() {
```
Comment (jgresty) [inline remove_elements.go:66]: This is a very narrow use case but this only covers a small part of the doc compared to the old implementation

File: resource.go [hunk 2]
```diff
@@ -331,14 +337,37 @@ func loadResource(specPath string, versionStr string) (*ResourceVersion, error)
...
-	doc.InternalizeRefs(ctx, nil)
+	doc.InternalizeRefs(ctx, ResolveRefsWithoutSourceName)
 	return doc.ResolveRefs()
 }
+
+// ResolveRefsWithoutSourceName resolves references without the source url/file name in ref
+// background: this was the way kin-openapi used to resolve references, but it was changed
+// in the recent versions(v0.127.0) to include the filename in the ref name. Although this
+// method prevents conflicts, it causes existing specs to break.
+func ResolveRefsWithoutSourceName(t *openapi3.T, componentRef openapi3.ComponentRef) string {
```
Comment (jgresty) [inline resource.go:359]: I can't clearly see what this is doing which implies some tests would be appreciated to use as examples

**Review by jgresty — APPROVED** [review]
This is a lot clearer than it was before however I don't really like the amount of repetitive code. I think there could be a middle ground with some light reflection since all of these types we are operating over have the same `.Ref` and `.Value` members which is all we care about.

Not going to block on that but it is an area we could improve on if we find this is a source of future bugs

File: remove_elements.go (hunk 1 above)
Comment (tinygrasshopper) [reply remove_elements.go:66]: agreed; I would argue the old implementation going much wider than intended. I was testing with the existing specs didnt find a use for replacing other extensions

**Pull request merged**